src/include/random.h

```diff
@@ -133,13 +133,14 @@ void randomize_rng()
   detail::randomize_rng<EngineT>();
 }
 
-template <int min = 0,
-          int max = std::numeric_limits<int>::max(), 
-          typename DistributionT = std::uniform_int_distribution<int>,
+template <typename IntegerT,
+          IntegerT min = std::numeric_limits<IntegerT>::min(),
+          IntegerT max = std::numeric_limits<IntegerT>::max(),
+          typename DistributionT = std::uniform_int_distribution<IntegerT>,
```

Comment:
@chardan how do you feel about this change? i was trying to use `generate_random_number<uint64_t>()` and found that this was only specialized for `int`. making this generic has the side effect that it can return negative numbers for `int` instead of starting at `int min = 0`. but i think it's more natural for `generate_random_number<T>()` to use the entire range of values available to `T`

Reply:
I'm of two minds here. I'm ok with allowing different integer types, but if we allow negatives ranges we'll have to update the tests as well. The reason the range is from 0 to max is because the original mission was to replace rand(), which provides a similar range. So the minimal change here might be to take a parametric type (default to int) and keep the 0 range; add a test that checks the parametric type works with a few different kinds of int. If we want to move from 0, there will need to probably be more checking, but I'm not opposed to it (just more work for you! ;->).

Reply:
okay. i think the defaults for some of these overloads are confusing as is - for example, `generate_random_number(5)` uses min=0 and max=5 despite 5 being the first argument

so i'd rather have the callers be explicit about the range they request, even if it means they have to mess with with `std::numeric_limits` themselves. i'll revert this change, and avoid the calls to `generate_random_number<uint64_t>()`

Reply:
@cbodley I'm ok with your change, just make IntegerT = int. Reverting is fine too, but I like what you did.